Assets/Scripts/Interaction/RobotInteractionHandler.cs

```diff
@@ -0,0 +1,34 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using Unity.VisualScripting.YamlDotNet.Core.Tokens;
+using UnityEngine;
+
+namespace Cosmobot
+{
+    public class RobotInteractionHandler : MonoBehaviour
+    {
+        private IInteractable interaction = null;
+        private void OnTriggerEnter(Collider other)
+        {
+            if (other.TryGetComponent(out IInteractable encounteredInteraction))
+            {
+                interaction = encounteredInteraction;
+            }
+        }
+        private void OnTriggerExit(Collider other)
+        {
+            if (other.TryGetComponent(out IInteractable encounteredInteraction))
+            {
+                if(encounteredInteraction==interaction)
+                {
+                    interaction=null;
+                }
+            }
+        }
+        public void Interact()
+        {
+            interaction.Use();
```

Comment:
Won't this NullRef?

Reply:
It was discussed with @PatrykPaluch and the conclusion was that it won't, because robots script will handle it.